sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -148,6 +148,54 @@
  *    <li>Since version: 3.3.0</li>
  *   </ul>
  *  </li>
+ *  <li>Name: <code>SUBSTRING</code>
+ *   <ul>
+ *    <li>SQL semantic: <code>SUBSTRING(string, position, length)</code></li>
```

Comment:
the length parameter here is also optional. We can exclude it if it's a int literal of `Int.MaxValue`